bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Table.java

```diff
@@ -86,16 +86,16 @@ public class Table extends Composite {
 	boolean shouldScroll = true;
 	boolean keyDown;
 
+	final int nativeItemHeight;
+	private boolean enforceNativeItemHeightMinimum;
```

Comment:
We would lose the ability to configure this setting individually, however unless we want setEnforceNativeItemHeightMinimum `public`, I think we could simplify this indeed.

Reply:
Indeed. If `enforceNativeItemHeightMinimum` is removed from Table/Tree/List and referenced like `display.enforceNativeItemHeightMinimum` then the `setEnforceNativeItemHeightMinimum` methods in Table/Tree/List becomes:
 
```java
void setEnforceNativeItemHeightMinimum() {
	setItemHeight(null, null, true); // update height
}
```

And maybe that method can be removed and `setItemHeight` called in the constructor?

Reply:
> We would lose the ability to configure this setting individuall

I don't think that's possible unless it becomes API which is not really suitable for platform-dependent.

Reply:
I had hoped to keep the method public. I don't quite understand the process in the Eclipse project to make something API, but you could of course totally have a public method that is only accessible for cocoa-specific code. Anyhow. I've now simplified the patch.

Reply:
If you develop solely on a Mac and see a public API method you might use it forgetting that applies to Mac only. It will compile and run, except when you try it on Windows or Linux...so all API methods should be cross-platform. I think that's right @lshanmug ?

Reply:
Ah, you're right. I forgot that this implementation shares the same package as on Linux/Windows. Nevermind.

Reply:
> If you develop solely on a Mac and see a public API method you might use it forgetting that applies to Mac only. It will compile and run, except when you try it on Windows or Linux...so all API methods should be cross-platform. I think that's right @lshanmug ?

That's right @Phillipus. Otherwise we will have to add empty stubs on other platforms to maintain code portability. So, when we create a new API we try to have implementations for atleast 2 platforms.